### PR TITLE
Replacing deprecated components and fixing prop errors

### LINKED ISSUE #19736: Fix FormField console error
### Description
There is an incorrect prop being passed to the FormField component causing a console error. 

https://github.com/MetaMask/metamask-extension/blob/365c1e32d262e45138b06f149dec81607359376b/ui/components/ui/form-field/form-field.js#L66

```
Warning: Received `true` for a non-boolean attribute `html`.
```

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation
Some prop type errors were introduced with some of the replacing of deprecated components. This PR replaces some deprecated components, removes the incorrect props and fixes some props being pass down to the html element in form-filed.js

- [Slack thread](https://consensys.slack.com/archives/GTQAGKY5V/p1687534783049359)
- [Slack thread](https://consensys.slack.com/archives/C0354T27M5M/p1687534404946349)
- Fixes https://github.com/MetaMask/metamask-extension/issues/19736
- Part of #18714 
- Part of #19526 
- Part of #18896 

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/8cbf2986-4f68-4ac5-b181-ab70f84d373c

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/7e0261ed-bf22-40b9-a49f-0678d5ab6bb7

## Manual Testing Steps
- Go to the latest build of storybook in this PR
- Search for `SecureYourWallet` 
- Check the story against develop story https://metamask.github.io/metamask-storybook/iframe.html?args=&id=pages-onboardingflow-secureyourwallet--default-story&viewMode=story

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/ui/form-field/form-field.js [hunk 1]
```diff
@@ -63,7 +63,6 @@ export default function FormField({
                 <Text
                   tag="label"
                   htmlFor={id}
-                  html
```
Comment (georgewrmarshall) [inline ui/components/ui/form-field/form-field.js:66]: Fixes incorrect prop being passed to Text component

File: ui/helpers/constants/design-system.ts [hunk 2]
```diff
@@ -215,13 +215,37 @@ export enum BorderStyle {
...
+   * 0
+   */
   none = 'none',
+  /**
+   * 9999px
+   */
   pill = 'pill',
+  /**
+   * 50%
+   */
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:248]: Not part of this issue but very helpful to see the pixel value when hovering over the size abstraction

Screenshot 2023-06-23 at 12 35 07 PM

File: ui/pages/onboarding-flow/secure-your-wallet/index.scss [hunk 3]
```diff
@@ -1,50 +1,10 @@
...
-
-  &__highlighted {
-    background-color: var(--color-primary-muted);
-    padding: 12px;
-    border-radius: 10px;
-
-    @include screen-sm-min {
-      margin-inline-end: 43px;
-    }
   }
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/secure-your-wallet/index.scss:8]: Removing unused CSS

File: ui/pages/onboarding-flow/secure-your-wallet/secure-your-wallet.js [hunk 4]
```diff
@@ -1,14 +1,17 @@
 import React, { useState, useContext } from 'react';
 import { useHistory, useLocation } from 'react-router-dom';
 import { useSelector } from 'react-redux';
-import Box from '../../../components/ui/box';
-import Button from '../../../components/ui/button';
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/secure-your-wallet/secure-your-wallet.js:5]: Replacing deprecated compoents

**Review by darkwing — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**